ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp

```diff
@@ -7324,6 +7324,309 @@ bool GDALGeoPackageDataset::DeleteVectorOrRasterLayer(const char *pszLayerName)
+                                 pszNewLayerName, pszLayerName);
+        osSQL += pszSQL;
+        sqlite3_free(pszSQL);
+
+        pszSQL = sqlite3_mprintf(
+            "UPDATE gpkg_2d_gridded_tile_ancillary SET tpudt_name = '%q' WHERE "
+            "lower(tpudt_name) = lower('%q');",
+            pszNewLayerName, pszLayerName);
+        osSQL += pszSQL;
+        sqlite3_free(pszSQL);
+    }
+
+    if (HasExtensionsTable())
+    {
+        pszSQL = sqlite3_mprintf(
+            "UPDATE gpkg_extensions SET table_name = '%q' WHERE "
+            "lower(table_name) = lower('%q');",
+            pszNewLayerName, pszLayerName);
+        osSQL += pszSQL;
+        sqlite3_free(pszSQL);
+    }
+
+    if (HasMetadataTables())
+    {
+        pszSQL = sqlite3_mprintf(
+            "UPDATE gpkg_metadata_reference SET table_name = '%q' WHERE "
+            "lower(table_name) = lower('%q');",
+            pszNewLayerName, pszLayerName);
+        osSQL += pszSQL;
+        sqlite3_free(pszSQL);
+    }
+
+    if (HasDataColumnsTable())
+    {
+        pszSQL = sqlite3_mprintf(
+            "UPDATE gpkg_data_columns SET table_name = '%q' WHERE "
+            "lower(table_name) = lower('%q');",
+            pszNewLayerName, pszLayerName);
+        osSQL += pszSQL;
+        sqlite3_free(pszSQL);
+    }
+
+    if (HasQGISLayerStyles())
+    {
+        // Update QGIS styles
+        pszSQL =
+            sqlite3_mprintf("UPDATE layer_styles SET f_table_name = '%q' WHERE "
+                            "lower(f_table_name) = lower('%q');",
+                            pszNewLayerName, pszLayerName);
+        osSQL += pszSQL;
+        sqlite3_free(pszSQL);
+    }
+
+#ifdef ENABLE_GPKG_OGR_CONTENTS
+    if (m_bHasGPKGOGRContents)
+    {
+        pszSQL = sqlite3_mprintf(
+            "UPDATE gpkg_ogr_contents SET table_name = '%q' WHERE "
+            "lower(table_name) = lower('%q');",
+            pszNewLayerName, pszLayerName);
+        osSQL += pszSQL;
+        sqlite3_free(pszSQL);
+    }
+#endif
+
+    if (HasGpkgextRelationsTable())
+    {
+        pszSQL = sqlite3_mprintf(
+            "UPDATE gpkgext_relations SET base_table_name = '%q' WHERE "
+            "lower(base_table_name) = lower('%q');",
+            pszNewLayerName, pszLayerName);
+        osSQL += pszSQL;
+        sqlite3_free(pszSQL);
+
+        pszSQL = sqlite3_mprintf(
+            "UPDATE gpkgext_relations SET related_table_name = '%q' WHERE "
+            "lower(related_table_name) = lower('%q');",
+            pszNewLayerName, pszLayerName);
+        osSQL += pszSQL;
+        sqlite3_free(pszSQL);
+
+        pszSQL = sqlite3_mprintf(
+            "UPDATE gpkgext_relations SET mapping_table_name = '%q' WHERE "
+            "lower(mapping_table_name) = lower('%q');",
+            pszNewLayerName, pszLayerName);
+        osSQL += pszSQL;
+        sqlite3_free(pszSQL);
+    }
+
+    // Drop all triggers for the layer
```

Comment:
Did you find dropping and recreating triggers needed ? Normally given the SQLite versions we target now, SQLite should normally update existing triggers:

Cf
```
$ sqlite3
SQLite version 3.31.1 2020-01-27 19:55:54
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> create table foo(id);
sqlite> CREATE TRIGGER my_trigger BEFORE INSERT ON foo FOR EACH ROW BEGIN SELECT RAISE(ABORT, 'error'); end;
sqlite> select * from sqlite_master;
table|foo|foo|2|CREATE TABLE foo(id)
trigger|my_trigger|foo|0|CREATE TRIGGER my_trigger BEFORE INSERT ON foo FOR EACH ROW BEGIN SELECT RAISE(ABORT, 'error'); end
sqlite> alter table foo rename to bar;
sqlite> select * from sqlite_master;
table|bar|bar|2|CREATE TABLE "bar"(id)
trigger|my_trigger|bar|0|CREATE TRIGGER my_trigger BEFORE INSERT ON "bar" FOR EACH ROW BEGIN SELECT RAISE(ABORT, 'error'); end
```

Reply:
I've found it necessary only because the name of the trigger itself is not changed, so the test   `assert "weird" not in content` fails at https://github.com/OSGeo/gdal/pull/10213/files#diff-cf1e29b5cee8a8c395aacdaba2a8b883b9f09e8f03beb9b24527ddea1eb7bb46R4350

Reply:
Sure, done.